token-validation-core/src/main/java/no/nav/security/token/support/core/validation/JwtTokenAnnotationHandler.java

```diff
@@ -77,25 +116,21 @@ protected Annotation getAnnotation(Method method, List<Class<? extends Annotatio
         return null;
     }
 
-    private static Annotation findAnnotation(Annotation[] annotations, List<Class<? extends Annotation>> types){
-        return annotations != null ?
-            Arrays.stream(annotations)
+    private static Annotation findAnnotation(Annotation[] annotations, List<Class<? extends Annotation>> types) {
+        return annotations != null ? Arrays.stream(annotations)
                 .filter(a -> types.contains(a.annotationType()))
                 .findFirst()
                 .orElse(null) : null;
     }
 
-    protected boolean handleProtectedWithClaimsAnnotation(ProtectedWithClaims annotation) {
-        return handleProtectedWithClaims(annotation.issuer(), annotation.claimMap(), annotation.combineWithOr());
+    protected boolean handleProtectedWithClaimsAnnotation(ProtectedWithClaims annotation, JwtToken jwtToken) {
+        return handleProtectedWithClaims(annotation.issuer(), annotation.claimMap(), annotation.combineWithOr(), jwtToken);
     }
 
-    protected boolean handleProtectedWithClaims(String issuer, String[] requiredClaims, boolean combineWithOr) {
+    protected boolean handleProtectedWithClaims(String issuer, String[] requiredClaims, boolean combineWithOr, JwtToken jwtToken) {
         if (Objects.nonNull(issuer) && issuer.length() > 0) {
-
-            JwtToken jwtToken = getJwtToken(issuer, tokenValidationContextHolder)
-                .orElseThrow(() -> new JwtTokenMissingException("no valid token found in validation context"));
             if (!containsRequiredClaims(jwtToken, combineWithOr, requiredClaims)) {
-                throw new JwtTokenInvalidClaimException("required claims not present in token." + Arrays.asList(requiredClaims));
+                return false;
```

Comment:
This change will result in losing context about the error such as which claims are missing. I understand that it has to be this way in order to loop through multiple issuers. Maybe we should look at introducing a return object `Result` containing true or false and any errors.

Reply:
scratch this. I see that this is handled in the `assertValidAnnotation` method